src/colormaps.jl

```diff
@@ -14,1053 +14,1387 @@
+                   (1.0, 0.0, 0.0),
+                   (0.803922, 0.0, 0.0)]
+
+const _snow_data = [(0.933333, 0.933333, 0.933333),
+                    (0.894118, 0.894118, 0.894118),
+                    (0.854902, 0.854902, 0.854902),
+                    (0.815686, 0.815686, 0.815686),
+                    (0.776471, 0.776471, 0.776471),
+                    (0.737255, 0.737255, 0.737255),
+                    (0.698039, 0.698039, 0.698039),
+                    (0.658824, 0.658824, 0.658824),
+                    (0.619608, 0.619608, 0.619608),
+                    (0.580392, 0.580392, 0.580392),
+                    (0.541176, 0.541176, 0.541176),
+                    (0.501961, 0.501961, 0.501961)]
+
+const _ash_data = [(0.462745, 0.462745, 0.462745),
+                   (0.423529, 0.423529, 0.423529),
+                   (0.384314, 0.384314, 0.384314),
+                   (0.345098, 0.345098, 0.345098),
+                   (0.305882, 0.305882, 0.305882),
+                   (0.266667, 0.266667, 0.266667),
+                   (0.227451, 0.227451, 0.227451),
+                   (0.188235, 0.188235, 0.188235),
+                   (0.14902, 0.14902, 0.14902),
+                   (0.109804, 0.109804, 0.109804),
+                   (0.070588, 0.070588, 0.070588),
+                   (0.031373, 0.031373, 0.031373)]
+
+const _gray_data = hcat(_snow_data, _ash_data)
+
+const _green_cyan_blue_data = [(0.0, 0.372549, 0.0),
+                               (0.0, 0.529412, 0.0),
+                               (0.0, 0.686275, 0.0),
+                               (0.0, 0.843137, 0.0),
+                               (0.0, 1.0, 0.0),
+                               (0.0, 1.0, 0.372549),
+                               (0.0, 1.0, 0.529412),
+                               (0.0, 1.0, 0.686275),
+                               (0.0, 1.0, 0.843137),
+                               (0.0, 1.0, 1.0),
+                               (0.0, 0.843137, 1.0),
+                               (0.0, 0.686275, 1.0),
+                               (0.0, 0.529412, 1.0),
+                               (0.0, 0.372549, 1.0),
+                               (0.0, 0.0, 1.0),
+                               (0.0, 0.0, 0.843137),
+                               (0.0, 0.0, 0.686275),
+                               (0.0, 0.0, 0.529412),
+                               (0.0, 0.0, 0.372549)]
+
+const COLOR_MAP_DATA = Dict(
     :viridis => _viridis_data,
     :inferno => _inferno_data,
     :magma   => _magma_data,
-    :plasma  => _plasma_data
+    :plasma  => _plasma_data,
+    :cividis  => _cividis_data,
+    :green_cyan_blue => _green_cyan_blue_data,
+    :gray => _gray_data,
+    :snow => _snow_data,
+    :ash => _ash_data,
+    :jet => _jet_data,
 )
 
-function closestansiicolor(c)
-    tc = [0, 95, 135, 175, 215, 255]
-    _, i = findmin(abs.(c .- tc))
-    i - 1
-end
+closestansiicolor(c) = argmin(abs.(c .- UInt8[0, 95, 135, 175, 215, 255])) - 1
```

Comment:
Performance suggestion:

```suggestion
const _closest_ansii_color_list = UInt8[0, 95, 135, 175, 215, 255]
if VERSION >= v"1.7"
    closestansiicolor(c) = argmin(v->abs(c-v), _closest_ansii_color_list) - 1
else
    closestansiicolor(c) = argmin(abs(c-v) for v in _closest_ansii_color_list) - 1
end
```

Reply:
I agree on the `const`.
Couldn't we use `closestansiicolor(c) = argmin(map(v -> abs(c - v), _closest_ansii_color_list)) - 1` for both versions ?

Reply:
`map` allocates memories; `map` almost work equivalently to `[abs(c - v) for v in ...]` and is slower than the generator version `(abs(c - v) for v in ...)`.

The v"1.7" version uses faster for-loop implementation and get rid of the generator overhead.

Reply:
Thanks, I tested it, wasn't aware of the map(...) allocation. On `1.6`, however this fails with `ERROR: MethodError: no method matching keys(::Base.Generator`

Reply:
I don't know why, but the 1.7 version `argmin(v->abs(c-v), _closest_ansii_color_list)` is completely broken and fails the CI.